src/miner.cpp

```diff
@@ -297,22 +423,29 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn)
                     dPriority, feeRate.ToString(), tx.GetHash().ToString());
             }
 
-            // Add transactions that depend on this one to the priority queue
-            if (mapDependers.count(hash))
+            bool fResort = false;
+            BOOST_FOREACH(CTxInfo* ptxinfo, vAdded)
             {
-                BOOST_FOREACH(COrphan* porphan, mapDependers[hash])
+                pblock->vtx.push_back(*ptxinfo->ptx);
+                pblocktemplate->vTxFees.push_back(ptxinfo->nTxFee);
+                pblocktemplate->vTxSigOps.push_back(ptxinfo->nTxSigOps);
+                nFees += ptxinfo->nTxFee;
+
+                ptxinfo->fInvalid = true;
+                if (!ptxinfo->setDependents.empty())
                 {
-                    if (!porphan->setDependsOn.empty())
+                    fResort = true;
```

Comment:
Aren't this !empty check and flag to resort redundant given the foreach loop below?